scanners/ncrack/parser/parser.js

```diff
@@ -0,0 +1,51 @@
+const xml2js = require('xml2js');
+
+async function parse(fileContent) {
+    const { ncrackrun } = await transformXML(fileContent);
+    const findings = transformToFindings(ncrackrun);
+    return findings;
+}
+
+function transformToFindings(ncrackrun) {
+    const portFindings = ncrackrun.service.flatMap(({ address, port, credentials = [] }) => {
+        const { addr: ipAddress } = address[0]['$'];
+        const { protocol, portid, name: portName } = port[0]['$'];
+
+        return credentials.map(credential => {
+            const { username, password } = credential['$'];
+
+            return {
+                name: `Credentials for Service ${portName}://${ipAddress}:${portid} discovered via bruteforce.`,
+                description: '',
+                category: 'Discovered Credentials',
+                location: `${portName}://${ipAddress}:${portid}`,
+                osi_layer: 'APPLICATION',
+                severity: 'HIGH',
+                attributes: {
+                    port: portid,
+                    ip_address: ipAddress,
+                    protocol: protocol,
+                    service: portName,
+                    username,
+                    password,
```

Comment:
I'm not sure if we should include the password.

Reply:
I think this needs to be in here because otherwise you can't really confirm that it was properly identified.
But its probably a good idea to write this in the docs for ncrack that users need to consider this.